LibraryApplication/LibraryApplication/EditBook.aspx.cs

```diff
@@ -0,0 +1,112 @@
+        private const string BooksUrl = "~/Books.aspx";
+        private const string GetBookQuery = @"
+            SELECT 
+                Book.Id AS BookId, 
+                Title, 
+                Author.Id AS AuthorId,
+                Author.FirstName + ' ' + Author.LastName AS AuthorName, 
+                Isbn
+            FROM Book JOIN Author
+            ON Book.AuthorId = Author.Id
+            WHERE Book.Id = @BookId
+        ";
+        private const string GetAuthorsQuery = @"
+            SELECT Id, LastName + ', ' + FirstName AS AuthorName
+            FROM Author
+            ORDER BY AuthorName
+        ";
+        private const string UpdateBookQuery = @"
+            UPDATE Book
+            SET
+                Title=@Title,
+                AuthorId=@AuthorId,
+                Isbn=@Isbn
+            WHERE Id=@BookId;
+        ";
+        private int bookId;
+
+        protected void Page_Load(object sender, EventArgs e)
+        {
+            if (!int.TryParse(Request.QueryString["ID"], out bookId))
+            {
+                Response.Redirect(BooksUrl);
+            }
+
+            if (!IsPostBack)
+            {
+                AuthorDropDown.ListDataSource = () => 
+                {
+                    DataTable authorsTable = DatabaseHelper.Retrieve(GetAuthorsQuery);
+                    AuthorDropDown.SetTextAndValueFields("AuthorName", "Id");
+                    return authorsTable;
+                };
+                DataTable bookTable = DatabaseHelper.Retrieve(
+                    GetBookQuery,
+                    new SqlParameter("@BookId", bookId));
+
+                if (bookTable.Rows.Count == 1)
+                {
+                    string oldTitle = bookTable.Rows[0].Field<string>("Title");
+                    string oldAuthor = bookTable.Rows[0].Field<string>("AuthorName");
+                    string oldIsbn = bookTable.Rows[0].Field<string>("Isbn");
+                    int oldAuthorId = bookTable.Rows[0].Field<int>("AuthorId");
+                    OldTitle.Text = oldTitle;
+                    OldAuthor.Text = oldAuthor;
+                    OldIsbn.Text = oldIsbn;
+                    TitleTextBox.Text = oldTitle;
+                    AuthorDropDown.SelectedValue = oldAuthorId.ToString();
+                    IsbnTextBox.Text = oldIsbn;
+                }
+                else
+                {
+                    Response.Redirect(BooksUrl);
+                }
+            }
+        }
+
+        protected void SaveButton_Command(object sender, CommandEventArgs e)
+        {
+            if (Page.IsValid)
+            {
+                string newTitle = TitleTextBox.Text;
+                int newAuthorId = int.Parse(AuthorDropDown.SelectedValue);
+
+                // Isbn is nullable in the database
+                string isbn = IsbnTextBox.Text;
+                if (string.IsNullOrWhiteSpace(isbn))
+                {
+                    isbn = null;
+                }
```

Comment:
The `GetNullableStringSqlParameter` method handles empty strings, nulls, and white space, so you don't have to force empty strings to nulls.